### PR TITLE
REP-45 Add option to delete replication data through UI

### LINKED ISSUE #45: Add support for more clean up options when deleting Replications
**Describe the task**
Add support for better clean up options when deleting Replications from the UI. Currently the CLI allows deleting a Replication's history and replicated items in addition to deleting the configuration itself. This should be possible from the UI as well.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by peterhuffer**
#### What does this PR do?
Closed https://github.com/connexta/replication/pull/52 to open against 0.2.x. I will apply comments from that PR.

- Fixes an issue in the UI where clicking the save configuration button twice quickly would save 2 configurations with the same name
- Adds option to delete replicated data when deleting a configuration. Only data replicated from another node to the local node will be deleted.
- Add a scheduled deleter that checks for deleted configurations and cleans them up (runs every minute)
- Add checks for deleting orphaned replication items (replication items without corresponding configurations or metacards).

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@clockard @paouelle @kcover @mcalcote 

#### How should this be tested? (List steps with links to updated documentation)
Make sure `/solr` can be accessed to check replication items.

- Replicate and verify deleting data along with the replication.
- Replicate and delete a config (preserve data). Delete a metacard and verify its corresponding replication item entry gets cleaned up.

#### What are the relevant tickets?
Fixes #45 

#### Screenshots (if appropriate)
image

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by peterhuffer** [comment]
Applied some changes for the comments on #52

**Comment by peterhuffer** [comment]
Pushed up changes for comments on #45

File: replication-api/src/main/java/org/codice/ditto/replication/api/data/ReplicatorConfig.java [hunk 1]
```diff
@@ -136,4 +136,37 @@
...
+   * See {@link #deleteData()}.
+   *
+   * @return whether or not this {@code ReplicatorConfig} should be considered as deleted
+   */
+  boolean isDeleted();
+
+  /**
+   * Marks this {@code ReplicatorConfig} as deleted.
+   *
+   * @param deleted
```
Comment (mcalcote) [inline replication-api/src/main/java/org/codice/ditto/replication/api/data/ReplicatorConfig.java]: ✏️ Other params have a description

File: replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ScheduledReplicatorDeleter.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+      Security security,
+      long pollPeriod,
+      int pageSize) {
+    this.replicatorConfigManager = replicatorConfigManager;
+    this.scheduledExecutorService = scheduledExecutorService;
+    this.replicationItemManager = replicationItemManager;
+    this.replicatorHistory = replicatorHistory;
+    this.metacards = metacards;
+    this.security = security;
+    this.pageSize = pageSize;
```
[the remark this answers is not in the document]
Comment (peterhuffer) [reply replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ScheduledReplicatorDeleter.java]: Ya for testing.

[the remark this answers is not in the document]
Comment (paouelle) [reply replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ScheduledReplicatorDeleter.java]: ✏️ Annotate with `@VisibleForTesting` in this case

File: replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/data/ReplicatorConfigImpl.java [hunk 3]
```diff
@@ -173,4 +194,24 @@ public boolean isSuspended() {
...
+    return deleted;
+  }
+
+  @Override
+  public void setDeleted(boolean deleted) {
+    this.deleted = deleted;
+  }
+
+  @Override
+  public boolean deleteData() {
```
Comment (paouelle) [inline replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/data/ReplicatorConfigImpl.java]: ✏️ `shouldDeleteDate()`

File: replication-api-impl/src/test/java/org/codice/ditto/replication/api/impl/MetacardsTest.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class MetacardsTest {
+
+  private Metacards metacards;
+
+  @Mock CatalogFramework catalogFramework;
+
+  @Before
+  public void setup() {
```
Comment (paouelle) [inline replication-api-impl/src/test/java/org/codice/ditto/replication/api/impl/MetacardsTest.java]: ❓ Why use a setup method her as opposed to just putting this in the attribute initialization above?

File: replication-api/src/main/java/org/codice/ditto/replication/api/data/ReplicatorConfig.java [hunk 5]
```diff
@@ -136,4 +136,37 @@
...
+  /**
+   * Applies only when {@link #isDeleted()} returns {@code true}.
+   *
+   * <p>Only data that has been replicated to this {@link ReplicationSite} from a remote {@link
+   * ReplicationSite} will be deleted.
+   *
+   * @return if {@code true}, delete the associated data replicated by this {@code
+   *     ReplicatorConfig}, otherwise retain the data.
+   */
+  boolean deleteData();
```
Comment (paouelle) [inline replication-api/src/main/java/org/codice/ditto/replication/api/data/ReplicatorConfig.java]: ✏️ `shouldDeleteData` would make it clearer in the conditions where it ends up being used

File: replication-api-impl/src/test/java/org/codice/ditto/replication/api/impl/MetacardsTest.java (hunk 4 above)
Comment (peterhuffer) [reply replication-api-impl/src/test/java/org/codice/ditto/replication/api/impl/MetacardsTest.java]: No reason in particular.

File: ui/src/main/webapp/components/Navbar.js [hunk 6]
```diff
@@ -43,7 +43,7 @@ const HelpDialog = props => {
   return (
     <Dialog open={open} onClose={handleClose} maxWidth='md'>
       <DialogTitle variant='h6' id='help-dialog-title'>
-        {'Welcome to the Project Charleston BETA'}
```
Comment (clockard) [inline ui/src/main/webapp/components/Navbar.js]: This removes it from the welcome dialog but not the navbar at the top of the page. Looks like there is one more removal needed

**Comment by clockard** [comment]
Manually verified functionality ✅

File: ui/src/main/webapp/components/Navbar.js (hunk 6 above)
Comment (peterhuffer) [reply ui/src/main/webapp/components/Navbar.js]: Whoops, removed.

**Pull request merged**